code/modules/reagents/chemistry/machinery/chem_heater.dm

```diff
@@ -137,20 +150,26 @@
 	// update the ui if it exists, returns null if no ui is passed/found
 	ui = SSnanoui.try_update_ui(user, src, ui_key, ui)
 	if(!ui)
-		ui = new(user, src, ui_key, "chem_heater.tmpl", "ChemHeater", 350, 270)
+		ui = new(user, src, ui_key, "chem_heater.tmpl", "ChemHeater", 350, 290)
 		ui.open()
 
 /obj/machinery/chem_heater/ui_data(mob/user, ui_key = "main", datum/topic_state/state = GLOB.default_state)
 	var/data[0]
+	var/cur_temp = beaker ? beaker.reagents.chem_temp : null
 
 	data["targetTemp"] = desired_temp
+	data["targetTempReached"] = ""
+	data["autoEject"] = auto_eject
 	data["isActive"] = on
 	data["isBeakerLoaded"] = beaker ? 1 : 0
 
-	data["currentTemp"] = beaker ? beaker.reagents.chem_temp : null
+	data["currentTemp"] = cur_temp
 	data["beakerCurrentVolume"] = beaker ? beaker.reagents.total_volume : null
 	data["beakerMaxVolume"] = beaker ? beaker.volume : null
 
+	if (cur_temp)
+		data["targetTempReached"] = round(cur_temp) == round(desired_temp) ? "good" : "average"
```

Comment:
key/variable name suggests that the value will be a boolean yet is a classname. Best to change it to a boolean for readability in the UI side.

Reply:
`targetTempState` would have probably been a more suitable key in hindsight, but I changed the value type to a boolean as per your previous comment to make the template code easier to read, using a ternary operator.